handlers.go

```diff
@@ -0,0 +1,247 @@
+package firecracker
+
+import (
+	"context"
+	"fmt"
+)
+
+// Handler name constants
+const (
+	StartVMMHandlerName                = "fcinit.StartVMM"
+	BootstrapLoggingHandlerName        = "fcinit.BootstrapLogging"
+	CreateMachineHandlerName           = "fcinit.CreateMachine"
+	CreateBootSourceHandlerName        = "fcinit.CreateBootSource"
+	AttachDrivesHandlerName            = "fcinit.AttachDrives"
+	CreateNetworkInterfacesHandlerName = "fcinit.CreateNetworkInterfaces"
+	AddVsocksHandlerName               = "fcinit.AddVsocks"
+	SetMetadataHandlerName             = "fcinit.SetMetadata"
+
+	ValidateCfgHandlerName = "validate.Cfg"
+)
+
+// StartVMMHandler is a named handler that will handle starting of the VMM.
+// This handler will also set the exit channel on completion.
+var StartVMMHandler = Handler{
+	Name: StartVMMHandlerName,
+	Fn: func(ctx context.Context, m *Machine) error {
+		m.logger.Debugf(fmt.Sprintf("%s handler executing", StartVMMHandlerName))
+		return m.startVMM(ctx)
+	},
+}
+
+// BootstrapLoggingHandler is a named handler that will set up fifo logging of
+// firecracker process.
+var BootstrapLoggingHandler = Handler{
+	Name: BootstrapLoggingHandlerName,
+	Fn: func(ctx context.Context, m *Machine) error {
+		m.logger.Debugf(fmt.Sprintf("%s handler executing", BootstrapLoggingHandlerName))
+		if err := m.setupLogging(ctx); err != nil {
+			m.logger.Warnf("setupLogging() returned %s. Continuing anyway.", err)
+		} else {
+			m.logger.Debugf("back from setupLogging")
```

Comment:
nit: I think this could be more descriptive, maybe "setup logging: success" or something since it means err == nil?